src/Core/Domain/Product/QueryResult/SpecificPriceForEditing.php

```diff
@@ -0,0 +1,297 @@
+     */
+    private $shopGroupId;
+
+    /**
+     * @var int|null
+     */
+    private $shopId;
+
+    /**
+     * @var int|null
+     */
+    private $cartId;
+
+    /**
+     * @var int|null
+     */
+    private $currencyId;
+
+    /**
+     * @var int|null
+     */
+    private $catalogPriceRuleId;
+
+    /**
+     * @var int|null
+     */
+    private $countryId;
+
+    /**
+     * @var int|null
+     */
+    private $groupId;
+
+    /**
+     * @var int|null
+     */
+    private $customerId;
+
+    /**
+     * @var DateTime|null
+     */
+    private $dateTimeFrom;
+
+    /**
+     * @var DateTime|null
+     */
+    private $dateTimeTo;
+
+    /**
+     * @param int $specificPriceId
+     * @param string $reductionType
+     * @param DecimalNumber $reductionAmount
+     * @param bool $includeTax
+     * @param DecimalNumber $price
+     * @param int $fromQuantity
+     * @param int|null $shopGroupId
+     * @param int|null $shopId
+     * @param int|null $cartId
+     * @param int|null $currencyId
+     * @param int|null $catalogPriceRuleId
+     * @param int|null $countryId
+     * @param int|null $groupId
+     * @param int|null $customerId
+     * @param DateTime|null $dateTimeFrom
+     * @param DateTime|null $dateTimeTo
+     */
+    public function __construct(
+        int $specificPriceId,
+        string $reductionType,
+        DecimalNumber $reductionAmount,
+        bool $includeTax,
+        DecimalNumber $price,
+        int $fromQuantity,
+        ?DateTime $dateTimeFrom,
+        ?DateTime $dateTimeTo,
+        ?int $shopGroupId,
+        ?int $shopId,
+        ?int $cartId,
+        ?int $currencyId,
+        ?int $catalogPriceRuleId,
+        ?int $countryId,
+        ?int $groupId,
+        ?int $customerId
+    ) {
+        $this->specificPriceId = $specificPriceId;
+        $this->reductionType = $reductionType;
+        $this->reductionAmount = $reductionAmount;
+        $this->includeTax = $includeTax;
+        $this->price = $price;
+        $this->fromQuantity = $fromQuantity;
+        $this->shopGroupId = $shopGroupId;
+        $this->shopId = $shopId;
+        $this->cartId = $cartId;
+        $this->currencyId = $currencyId;
+        $this->catalogPriceRuleId = $catalogPriceRuleId;
+        $this->countryId = $countryId;
+        $this->groupId = $groupId;
+        $this->customerId = $customerId;
+        $this->dateTimeFrom = $dateTimeFrom;
+        $this->dateTimeTo = $dateTimeTo;
+    }
+
+    /**
+     * @return int
+     */
+    public function getSpecificPriceId(): int
+    {
+        return $this->specificPriceId;
+    }
+
+    /**
+     * @return string
+     */
+    public function getReductionType(): string
+    {
+        return $this->reductionType;
+    }
+
+    /**
+     * @return DecimalNumber
+     */
+    public function getReductionAmount(): DecimalNumber
+    {
+        return $this->reductionAmount;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isIncludeTax(): bool
```

Comment:
```suggestion
    public function doesItIncludeTax(): bool
```
Nice to have

Reply:
yeah these are debatable. (for sure `is` sounds bad)
I would go with simple `$includesTax`, `includesTax()`  (and even if we would need a setter I would call it `setIncludesTax(bool $includesTax)`
This way we dont have a problem with any propertyAccessor and it doesn't sound terrible :man_shrugging: